Nb walkthrough reformat (#20)
R4EPI/R4EPIs-website: content/training/Walk-through/clean_data.Rmd

nrow(linelist_cleaned %>% filter(is.na(date_of_consultation_admission))) 
```

{{% notice tip %}}
Notice that the command above returns the number of rows missing `date_of_consultation_admission`, but does not overwrite `linelist_cleaned` with an assignment operator `<-`. 
{{% /notice %}}

**After viewing the above results, we must comment out (#) the two lines that `filter()` the linelist_cleaned dataset - we do *not* want to remove those 616 observations.**

```{r}

> #linelist_cleaned <- linelist_cleaned %>% 

Reviewer: it's best practice to have a space after the hash

```suggestion
# linelist_cleaned <- linelist_cleaned %>% 
```

> #  filter(!is.na(case_number) & !is.na(date_of_consultation_admission)) 

Reviewer: ```suggestion
#   filter(!is.na(case_number) & !is.na(date_of_consultation_admission)) 
```

  ))
```

### Factor variables  

Factor variables require specialized functions to manipulate (see Factor sections of the Advanced and Miscellaneous R Basics page). The one factor variable we should clean now is `sex`, with the following code:  

```{r}
linelist_cleaned$sex <- fct_recode(linelist_cleaned$sex, 
                                   NULL = "Unknown/unspecified")
```                                   
                                   
                                   
**We can comment out (#) the rest of the `factor_vars` chunk**, such as the code to change the order of levels in categorical variables. The Am Timan dataset does not include a variable `time_to_death` and we do not need to change the order of any categorical variables.  

### Cleaning patient origin  

The variable `patient_origin` (originally the variable *quartier* in the raw dataset) is referenced in the place analyses chunks, for example when calculating attack rates by region. In those steps, the variable `patient_origin` in `linelist_cleaned` is *joined to* the variable `patient_origin` of the data frame `population_data_region` (which was imported in the population and lab data steps).  

> However, the place names in `population_data_region` are ALL CAPITAL LETTERS. This is not true in `linelist_cleaned` - so the join will not register any matches. The easiest way to solve this problem is to **redefine the `linelist_cleaned` variable into all capital letters as well**, using the base R function `toupper()`:  

Reviewer: ```suggestion
However, the place names in `population_data_region` are ALL CAPITAL LETTERS. This is not true in `linelist_cleaned` - so the join will not register any matches. The easiest way to solve this problem is to **redefine the `linelist_cleaned` variable into all capital letters as well**, using the base R function `toupper()`, which means "to upper case":  
```

```{r}
linelist_cleaned$patient_origin <- toupper(linelist_cleaned$patient_origin)
```

> ### Create list variables  

Reviewer: This title bothers me a bit because lists are a thing in R that are different from vectors. I don't think my solution is perfect, but it's perhaps a start?

```suggestion
### Gather common variables for later analysis
```

These variables names are stored in vectors that are created using the function `c()`. These vectors of names will be referenced in later code chunks. This way if you want to run the same function over these variables you can simply use the named vector rather than typing out each variable individually.  

> The default template code creates two vectors - one for symptoms variables (`SYMPTOMS`) and one for laboratory testing variables (`LABS`). **Some variables do not apply to our Am Timan data and must be removed from the list or commented out (such as `convulsions` and `history of fever`).** If non-existant variables are left in the list it will cause errors in later code chunks.

Reviewer: We've updated the tab_linelist code to only give warnings if variables don't exist (see https://github.com/R4EPI/sitrep/issues/183)

```suggestion
The default template code creates two vectors - one for symptom variables (`SYMPTOMS`) and one for laboratory testing variables (`LABS`). 
```
